lib/elixir/src/elixir_errors.erl

```diff
@@ -12,6 +12,12 @@
 warn(none, File, Warning) ->
   warn(0, File, Warning);
 warn(Line, File, Warning) when is_integer(Line), is_binary(File) ->
+  CompilerPid = get(elixir_compiler_pid),
+  if
+    CompilerPid =/= undefined ->
+      CompilerPid ! {warning, File, Line, list_to_binary(Warning)};
```

Comment:
`Warning` is `unicode:chardata` so we need `unicode:characters_to_binary(Warning)`, `list_to_binary/1` would only be for `latin1`.